### ISSUE TITLE
Attribute network requests to frames

### ISSUE THREAD

**Issue by aslushnikov** [puppeteer/puppeteer#1579]
We should attribute network requests to frames so that we know where requests are coming from.

```js
const frame = request.frame();
```

This would simplify network traffic analysis.

Open question: it might make sense to expose execution contexts instead to attribute requests coming from workers.

**Comment by yujiosaka** [comment]
+1

Use case: I was writing a patch to disable/relax CSP by reading a part from response headers, but it was hard to follow frame -> request -> response. It would be a nice feature to easily access the property like that.
